src/renderer/model/Model.cpp

```diff
@@ -0,0 +1,195 @@
+#include <rg/renderer/model/Model.hpp>
+
+#include <rg/renderer/model/Texture.hpp>
+#include <rg/renderer/model/Vertex.hpp>
+
+#include <glad/glad.h>
+
+#include <assimp/Importer.hpp>
+#include <assimp/postprocess.h>
+#include <assimp/scene.h>
+
+#include <spdlog/spdlog.h>
+
+#include <array>
+#include <memory>
+#include <unordered_map>
+#include <vector>
+
+namespace rg {
+
+void Model::draw(const Shader& shader) {
+    for (const auto& mesh : meshes_)
+        mesh.draw(shader);
+}
+
+namespace {
+
+class Loader {
+public:
+    explicit Loader(const std::string& path)
+            : meshes_{}, textures_{}, path_{path} {
+        directory_ = path.substr(0, path.find_last_of('/'));
+    }
+
+    void loadScene();
+    std::vector<Mesh> get_meshes() const {
+        return meshes_;
+    }
+
+private:
+    std::vector<Mesh> meshes_;
+    std::unordered_map<std::string, std::shared_ptr<Texture>> textures_;
+    std::string path_;
+    std::string directory_;
+
+    /** @brief Recursively processes the scene and converts it to our model
+     * format
+     *
+     * @param node The current node of the scene being processed
+     * @param scene Assimp's representation of a scene
+     * @param meshes An array to insert the meshes
+     * @param memo Used to prevent the same texture being loaded twice
+     */
+    void processNode(aiNode* node, const aiScene* scene);
+    Mesh processMesh(aiMesh* mesh, const aiScene* scene);
+    static Vertex processVertex(aiMesh* mesh, unsigned int index);
+
+    std::vector<std::shared_ptr<Texture>>
+    loadMaterialTextures(aiMaterial* material, TextureType type);
+};
+
+} // namespace
+
+Model::Model(const std::string& path) : meshes_{} {
+    Loader loader{path};
+    loader.loadScene();
+    meshes_ = loader.get_meshes();
+}
+
+namespace {
+
+void Loader::processNode(aiNode* node, const aiScene* scene) {
+    for (unsigned int i = 0; i < node->mNumMeshes; ++i) {
+        auto* mesh = scene->mMeshes[node->mMeshes[i]];
+        meshes_.emplace_back(processMesh(mesh, scene));
+    }
+
+    for (unsigned int i = 0; i < node->mNumChildren; ++i)
+        processNode(node->mChildren[i], scene);
+}
+
+Mesh Loader::processMesh(aiMesh* mesh, const aiScene* scene) {
+    std::vector<Vertex> vertices;
+    std::vector<unsigned int> indices;
+    std::vector<std::shared_ptr<Texture>> textures;
+
+    for (unsigned int i = 0; i < mesh->mNumVertices; ++i)
```

Comment:
Možda `vertices.reserve(mesh->mNumVertices);` pre petlje? Ne vidim da se to polje negde menja u processVertex

Reply:
```suggestion
    vertices.reserve(mesh->mNumVertices);
    for (unsigned int i = 0; i < mesh->mNumVertices; ++i)
```